### PR TITLE
Feat/token paymaster test refactor

### PR DESCRIPTION
**Pull request by livingrockrises**
- Tests are refactored to avoid stack too deep issues.
- Added maxPenalty logic in validation stage
- Updated type of tokenPrice var
- Added approve to router logic in constructor
- updated swap router and native address as per base mainnet
- added fork test for base mainnet

Note: some of the accounting tests are still unclear/unsatisfactory. 
Also, left some console logs (to be removed) for now

### CODE REVIEW

File: contracts/token/BiconomyTokenPaymaster.sol [hunk 1]
```diff
@@ -16,6 +16,8 @@ import { SignatureCheckerLib } from "solady/utils/SignatureCheckerLib.sol";
 import { ECDSA as ECDSA_solady } from "solady/utils/ECDSA.sol";
 import "account-abstraction/core/Helpers.sol";
 import "./swaps/Uniswapper.sol";
+// Todo: marked for removal
+import "forge-std/console2.sol";
```
Comment (filmakarov) [inline contracts/token/BiconomyTokenPaymaster.sol:20]: do you want to remove it after audit?

File: contracts/token/BiconomyTokenPaymaster.sol [hunk 2]
```diff
@@ -56,7 +58,8 @@ contract BiconomyTokenPaymaster is
     // supported in // independent mode
 
     // PAYMASTER_ID_OFFSET
-    uint256 private constant _UNACCOUNTED_GAS_LIMIT = 50_000; // Limit for unaccounted gas cost
+    // Note: Temp
```
Comment (filmakarov) [inline contracts/token/BiconomyTokenPaymaster.sol:61]: Is it for testing purposes?

File: contracts/token/BiconomyTokenPaymaster.sol [hunk 3]
```diff
@@ -123,6 +126,10 @@ contract BiconomyTokenPaymaster is
             independentTokenDirectory[independentTokensArg[i]] =
                 TokenInfo(oraclesArg[i], 10 ** IERC20Metadata(independentTokensArg[i]).decimals());
         }
+        // Approve swappable tokens for max amount
+        for (uint256 i = 0; i < swappableTokens.length; i++) {
```
Comment (filmakarov) [inline contracts/token/BiconomyTokenPaymaster.sol]:
```suggestion
        uint256 length = swappableTokens.length;
        for (uint256 i; i < length; i++) {
```

**Review by filmakarov — APPROVED** [review]

File: contracts/token/BiconomyTokenPaymaster.sol (hunk 1 above)
Comment (livingrockrises) [reply contracts/token/BiconomyTokenPaymaster.sol:20]: yes I will

File: contracts/token/BiconomyTokenPaymaster.sol (hunk 2 above)
Comment (livingrockrises) [reply contracts/token/BiconomyTokenPaymaster.sol:61]: yes, haven't gotten to right accounting yet.

**Pull request merged**